calico/felix/devices.py

```diff
@@ -175,15 +175,22 @@ def set_routes(ip_type, ips, interface, mac=None):
     :param set ips: IPs to set up (any not in the set are removed)
     :param str interface: Interface name
     :param str mac|NoneType: MAC address. May not be none unless ips is empty.
+    :param bool reset_arp: Reset arp. Only valid if IPv4.
     """
     if mac is None and ips:
         raise ValueError("mac must be supplied if ips is not empty")
+    if reset_arp and ip_type != futils.IPV4:
+        raise ValueError("reset_arp may only be supplied for IPv4")
 
     current_ips = list_interface_ips(ip_type, interface)
+
     for ip in (current_ips - ips):
         del_route(ip_type, ip, interface)
     for ip in (ips - current_ips):
         add_route(ip_type, ip, interface, mac)
+    if reset_arp and ip_type == futils.IPV4:
```

Comment:
Why do we need to check `ip_type == futils.IPV4`? Either `reset_arp` is false, so this second condition doesn't matter, or it's true, and then we know the second condition is true, or we'd have raised an exception on line 183.